apps/omg_watcher/test/exit_processor/core_test.exs

```diff
@@ -766,6 +806,330 @@ defmodule OMG.Watcher.ExitProcessor.CoreTest do
+        }
+        |> Core.find_ifes_in_blocks(state)
+
+      # note that outputs are not double-spent and can't be challenged; they can't exit since
+      # IFE is not canonical
+      assert {:ok, [%Event.InvalidPiggyback{txbytes: ^txbytes, inputs: [0], outputs: []}]} =
+               invalid_exits_filtered(request, state, only: [Event.InvalidPiggyback])
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "produces single challenge proof on double-spent piggyback input",
+         %{
+           invalid_piggyback_on_input: %{
+             state: state,
+             request: request,
+             bad_pb_output: bad_pb_output,
+             ife_txbytes: txbytes
+           }
+         } do
+      assert {:ok, [%{in_flight_input_index: ^bad_pb_output}]} =
+               Core.get_input_challenge_data(request, state, txbytes, bad_pb_output)
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for wrong oindex",
+         %{
+           invalid_piggyback_on_input: %{
+             state: state,
+             request: request,
+             bad_pb_output: bad_pb_output,
+             ife_txbytes: txbytes
+           }
+         } do
+      assert bad_pb_output != 1
+      assert {:error, :no_double_spent_on_particular_input} = Core.get_input_challenge_data(request, state, txbytes, 1)
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for wrong txhash",
+         %{invalid_piggyback_on_input: %{state: state, request: request}, competing_transactions: [_, _, comp3 | _]} do
+      comp3_txbytes = Transaction.encode(comp3)
+      assert {:error, :unknown_ife} = Core.get_input_challenge_data(request, state, comp3_txbytes, 0)
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for wrong badly encoded tx",
+         %{invalid_piggyback_on_input: %{state: state, request: request}, competing_transactions: [_, _, comp3 | _]} do
+      corrupted_txbytes = "corruption" <> Transaction.encode(comp3)
+      assert {:error, :malformed_transaction_rlp} = Core.get_input_challenge_data(request, state, corrupted_txbytes, 0)
+    end
+
+    @tag fixtures: [:invalid_piggyback_on_input, :in_flight_exits, :competing_transactions]
+    test "fail when asked to produce proof for illegal oindex",
+         %{invalid_piggyback_on_input: %{state: state, request: request, ife_txbytes: txbytes}} do
+      assert {:error, :no_double_spent_on_particular_input} = Core.get_input_challenge_data(request, state, txbytes, -1)
+    end
+
+    @tag fixtures: [:alice, :bob, :processor_filled, :transactions, :in_flight_exits, :competing_transactions]
+    test "can produce the multiple challenge proofs for a input piggyback",
```

Comment:
why do we support this? any challenge proof will do, and the API will server only one. Looks like unnecessary complexity

Reply:
fixed